cvmfs/sync_mediator.h

```diff
@@ -67,6 +67,38 @@ struct HardlinkGroup {
   FileChunkList file_chunks;
 };
 
+struct Counters {
```

Comment:
If the `Counters` struct isn't needed outside of the `SyncMediator` class, I would move it into `sync_mediator.cc` in a unnamed namespace and use a forward declaration here, to allow declaring the UniquePtr.